src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -696,6 +697,142 @@ class InsertStatementSegment(BaseSegment):
     )
 
 
+class BulkInsertStatementSegment(BaseSegment):
+    """A `BULK INSERT` statement.
+
+    https://learn.microsoft.com/en-us/sql/t-sql/statements/bulk-insert-transact-sql?view=sql-server-ver16
+    """
+
+    type = "bulk_insert_statement"
+    match_grammar = Sequence(
+        "BULK",
+        "INSERT",
+        Ref("TableReferenceSegment"),
+        "FROM",
+        Ref("QuotedLiteralSegment"),
+        Ref("BulkInsertStatementWithSegment", optional=True),
+    )
+
+
+class BulkInsertStatementWithSegment(BaseSegment):
+    """A `WITH` segment in the BULK INSERT statement.
+
+    https://learn.microsoft.com/en-us/sql/t-sql/statements/bulk-insert-transact-sql?view=sql-server-ver16
+    """
+
+    type = "bulk_insert_with_segment"
+    match_grammar = Sequence(
+        "WITH",
+        Bracketed(
+            Delimited(
+                AnyNumberOf(
+                    Sequence(
+                        "BATCHSIZE",
+                        Ref("EqualsSegment"),
+                        Ref("NumericLiteralSegment"),
+                    ),
+                    "CHECK_CONSTRAINTS",
+                    Sequence(
+                        "CODEPAGE",
+                        Ref("EqualsSegment"),
+                        Ref("QuotedLiteralSegment"),
+                    ),
+                    Sequence(
+                        "DATAFILETYPE",
+                        Ref("EqualsSegment"),
+                        Ref("QuotedLiteralSegment"),
+                    ),
+                    Sequence(
+                        "DATA_SOURCE",
+                        Ref("EqualsSegment"),
+                        Ref("QuotedLiteralSegment"),
+                    ),
+                    Sequence(
+                        "ERRORFILE",
+                        Ref("EqualsSegment"),
+                        Ref("QuotedLiteralSegment"),
+                    ),
+                    Sequence(
+                        "ERRORFILE_DATA_SOURCE",
+                        Ref("EqualsSegment"),
+                        Ref("QuotedLiteralSegment"),
+                    ),
+                    Sequence(
```

Comment:
A lot of these sequences are more or less the same, can you clean this up a bit? For ones with the same equals, can you stick a list of keywords wrapped in a OneOf()

Reply:
@WittierDinosaur done ✅